config/serverless-operator.yaml

```diff
@@ -71,3 +65,191 @@ repositories:
         requests:
           cpu: 100m
           memory: 200Mi
+    customConfigs:
+      - name: "aws-ovn" # Results in openshift-knative-serverless-operator-<BRANCH>__<OCP_VERSION>-aws-ovn.yaml
+        releaseBuildConfiguration:
+          tests:
+            - as: e2e-aws-ovn-continuous
+              cron: 0 0 * * 0
+              steps:
+                cluster_profile: aws
+                test:
+                  - as: serverless-e2e
+                    commands: make install-mesh test-e2e-with-kafka
+                    # dependencies: Generated by openshift-knative/hack
+                    from: serverless-source-image
+                    resources:
+                      limits:
+                        memory: 6Gi
+                      requests:
+                        cpu: 100m
+                        memory: 200Mi
+                workflow: openshift-e2e-aws-ovn
+      - name: "azure" # Results in openshift-knative-serverless-operator-<BRANCH>__<OCP_VERSION>-azure.yaml
+        releaseBuildConfiguration:
+          tests:
+            - as: e2e-azure-continuous
+              cron: 0 0 * * 0
+              steps:
+                cluster_profile: azure4
+                test:
+                  - as: serverless-e2e
+                    commands: make install-mesh test-e2e-with-kafka
+                    # dependencies: Generated by openshift-knative/hack
+                    from: serverless-source-image
+                    resources:
+                      limits:
+                        memory: 6Gi
+                      requests:
+                        cpu: 100m
+                        memory: 200Mi
+                workflow: openshift-e2e-azure
+      - name: "gcp" # Results in openshift-knative-serverless-operator-<BRANCH>__<OCP_VERSION>-gcp.yaml
+        releaseBuildConfiguration:
+          tests:
+            - as: e2e-gcp-continuous
+              cron: 0 0 * * 0
+              steps:
+                cluster_profile: gcp
+                test:
+                  - as: serverless-e2e
+                    commands: make install-mesh test-e2e-with-kafka
+                    # dependencies: Generated by openshift-knative/hack
+                    from: serverless-source-image
+                    resources:
+                      limits:
+                        memory: 6Gi
+                      requests:
+                        cpu: 100m
+                        memory: 200Mi
+                workflow: openshift-e2e-gcp
+      - name: "hypershift" # Results in openshift-knative-serverless-operator-<BRANCH>__<OCP_VERSION>-hypershift.yaml
+        releaseBuildConfiguration:
+          base_images:
+            hypershift-operator:
+              name: "4.13"
```

Comment:
This is one of the few places where the version is hardcoded in the custom config. I'm not sure how to do it better. We could maybe add these base images by default to all configs and have the version generated. It shouldn't harm.

Reply:
One possibility would be moving this whole `customConfigs` section under openshift-knative/serverless-operator and have the version properly generated there. But it would complicate the whole workflow - there would be two rounds of generating for the CI config (one in s-o, and then one in hack).
It might be alright to keep it where it is now and update the version in a few places. After all, It's just in one file.

Reply:
Another approach could be to special case serverless-operator in the code itself and inject all these flavor tests for SO only but we would base the various versions on the actual OCP version we're iterating on

Reply:
Another option could be having the custom config as a Go template. On this line above `name: {  .OCPVersion }`. And then we could fill in the template before parsing the configuration.

Reply:
> Another approach could be to special case serverless-operator in the code itself and inject all these flavor tests for SO only but we would base the various versions on the actual OCP version we're iterating on

I like that suggestion.